configure

```diff
@@ -310,61 +301,26 @@ if test "x$host" = 'xauto'; then
 fi
 
 
-# set arch from host
+# Add extra options for some hosts
 
-arch='none'
 case $host in
     x86_64-*linux-gnu|x86_64-kfreebsd-gnu)
-        ccdef="-DZ_ELF -DZ_DOT_LABEL_PREFIX $ccdef"
         if test $wordsize = 32; then
             # 32-bit OCaml on a 64-bit host
-            arch='i686'
             ccopt="-m32 $ccopt"
-            asopt="-m32 $asopt"
-        else
-            arch='x86_64'
         fi
         ;;
-    i486-*linux-gnu|i686-*linux-gnu|i486-kfreebsd-gnu)
-        ccdef="-DZ_ELF -DZ_DOT_LABEL_PREFIX $ccdef"
-        arch='i686';;
-    i686-*cygwin)
-        if test "x$wordsize" = "x64"; then
-            ccdef="-DZ_COFF $ccdef"
-            arch='x86_64_mingw64'
-        else
-            ccdef="-DZ_UNDERSCORE_PREFIX -DZ_COFF $ccdef"
-            arch='i686'
-        fi
-	;;
     i386-*darwin* | x86_64-*darwin*)
-        ccdef="-DZ_UNDERSCORE_PREFIX -DZ_MACOS $ccdef"
         if test "x$wordsize" = "x64"; then
             ccopt="-arch x86_64 $ccopt"
-            asopt="-arch x86_64 $asopt"
-            arch='x86_64'
             checkcc
         else
             ccopt="-arch i386 $ccopt"
-            asopt="-arch i386 $asopt"
-            arch='i686'
             checkcc
         fi
         ;;
-    armv7*-gnueabi)
-        arch='arm'
-        ;;
-    none)
-        ;;
-    *) 
-        echo "unknown host $host";;
 esac
 
```

Comment:
Note: this `switch` that sets `ccopt` in some cases may not be needed any longer.  Since we compile .c files via `ocamlc`, architecture-specific flags such as `-m32` or `-arch x86_64` are already provided by `ocamlc`.

Reply:
OK, I went ahead and removed this `case`.  Let's see what CI says, esp. on macOS.